### ISSUE TITLE
Rustc triggers BSoD on WSL1

### ISSUE THREAD

**Issue by arturkow2000** [microsoft/WSL#6383]
# Environment

```none
Windows build number: 10.0.21277.1000
Your Distribution version: Ubuntu 20.04
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.4.0-21277-Microsoft (Microsoft@Microsoft.com) (gcc version 5.4.0 (GCC) ) #1000-Microsoft Sun Dec 07 14:43:00 PST 2020
```

# Steps to reproduce
```bash
git clone --depth=1 https://gitlab.redox-os.org/redox-os/redox
cd redox
git submodule init
# module not required here, disabling greatly improves download time
git submodule deinit rust
git submodule update --recursive
make
```

# Actual behavior
While building relibc rustc triggers BSOD.

I will send relevant information to secure@microsoft.com

**Comment by Biswa96** [comment]
Which distribution? versions of the used softwares?

**Comment by arturkow2000** [comment]
Up-to-date Ubuntu 20.04
rustc 1.51.0-nightly (257becbfe 2020-12-27)
redox commit 5e39ab0609f81a3a4dc8d98f03b47e397f10ce83

**Comment by Biswa96** [comment]
Yes, I can reproduce your issue with WSL1 in Windows 10 build 21277 (Co release).  The LxCore.sys drive in 21277 gains 4 KB more than 20279. Have you tried with build 20279 (Fe release)?

**Comment by Biswa96** [comment]
Yes, I can reproduce your issue with WSL1 in Windows 10 build 21277 (Co release). I got stop code 'MEMORY MANAGEMENT'. The LxCore.sys drive in 21277 gains 4 KB more than 20279. Have you tried with build 20279 (Fe release)?

**Comment by arturkow2000** [comment]
> Yes, I can reproduce your issue with WSL1 in Windows 10 build 21277 (Co release). I got stop code 'MEMORY MANAGEMENT'. The LxCore.sys drive in 21277 gains 4 KB more than 20279. Have you tried with build 20279 (Fe release)?

No.

**Comment by DemiMarie** [comment]
Has this been fixed?

**Comment by DooferDarth** [comment]
I'm currently running Windows 11 Pro 64-bit Build 22000.469, and I'm able to consistently replicate this by running the official rustup install script for WSL on Ubuntu 20.04 WSL 1. I'm getting the same "MEMORY_MANAGEMENT" BSOD almost everytime I run it.

**Comment by DesktopECHO** [comment]
The BDOS is fixed in Windows 11 22H2 - Windows Server 2022 is still affected.
